kolibri/core/device/api.py

```diff
@@ -193,34 +197,67 @@ class Meta:
         fields = ["user", "member_of"]
 
 
+RECENTLY_SYNCED = "RECENTLY_SYNCED"
+SYNCING = "SYNCING"
+QUEUED = "QUEUED"
+NOT_RECENTLY_SYNCED = "NOT_RECENTLY_SYNCED"
+
+
+def map_status(status):
+    """
+    Summarize the current state of the sync into a constant for use by
+    the frontend.
+    """
+    if status["active"]:
+        return SYNCING
+    elif status["queued"]:
+        return QUEUED
+    elif status["last_synced"]:
+        if timezone.now() - status["last_synced"] < timedelta(hours=1):
+            return RECENTLY_SYNCED
+        else:
+            return NOT_RECENTLY_SYNCED
+
+
 class UserSyncStatusViewSet(ReadOnlyValuesViewset):
     permission_classes = (KolibriAuthPermissions,)
     filter_backends = (KolibriAuthPermissionsFilter, DjangoFilterBackend)
     queryset = UserSyncStatus.objects.all()
     filter_class = SyncStatusFilter
 
     values = (
-        "id",
         "queued",
         "last_synced",
         "active",
         "user",
-        "user_id",
     )
 
+    field_map = {
+        "status": map_status,
+    }
+
     def get_queryset(self):
-        return UserSyncStatus.objects.filter()
+        # If this is a subset of users device, we should just return no data
+        # if there are no possible devices we could sync to.
```

Comment:
Does this then make it show no status at all? Would we not want it to show a "not connected" message in this case?

Reply:
No - absence of status is interpreted by the frontend as 'not connected', so it does indeed do this!